### ISSUE TITLE
Questions about GPU and VRAM usage for Baseline finetune.

### ISSUE THREAD

**Issue by ChonghaoSima** [OpenDriveLab/DriveLM#57]
### Discussed in https://github.com/OpenDriveLab/DriveLM/discussions/56

<div type='discussions-op-text'>

<sup>Originally posted by **piqiuni** March 22, 2024</sup>
In the Table in [Finetune](https://github.com/OpenDriveLab/DriveLM/tree/main/challenge#setup), it says when Batch size = 4, the required VRAM is 34GB.
We are using RTX4090*2 for finetune, but we can only run with Batch size = 1, and the VRAM usage is about 20GB per card. Why it take so much VRAM?

Do you have any suggestions on the GPU usage? Should we try to get more GPUs for training? Maybe give me some detailed suggestions on GPU types and numbers?

Thanks for your suggestions! Forgive me for being a beginner~</div>

**Comment by piqiuni** [comment]
And in the [Inference baseline](https://github.com/OpenDriveLab/DriveLM/tree/main/challenge#inference-baseline), the demo set "--batch_size 4", but I got an assert in "challenge/llama_adapter_v2_multimodal7b/llama/llama_adapter.py" with "max_batch_size=1" in init of LLaMA_adapter. Can I change the "max_batch_size" in LLaMA_adapter.__init__? What influence it have?

Meanwhile, I think the "--num_processes" in  [Inference baseline](https://github.com/OpenDriveLab/DriveLM/tree/main/challenge#inference-baseline) should be the parallel process nums in each single GPU, not the ”# num_processes is the number of the gpu you will use to infer the data.“.

**Comment by DevLinyan** [comment]
1. VRAM consumption is influenced by both data and model size. While batch size primarily pertains to data, initializing the model also incurs a VRAM cost. Therefore, it's typical for a batch size of 1 to require around 20GB of VRAM. Increasing the batch size to 2 can provide insight into VRAM utilization. Personally, I find the 3090 sufficient, but training on an A100 (with 80GB) would offer even greater capability.

2. You can adjust the `max_batch_size`. It won't have any impact.

3. Within the code, you'll notice that num_processes can refer to either the parallel processes or the number of GPUs. They are equivalent.

**Comment by DevLinyan** [comment]
By the way, if your machine lacks sufficient memory, you can employ several methods to reduce VRAM usage. Here are some straightforward techniques:

1. Utilize gradient checkpointing.
2. Utilize optimized repositories like flash-attention.
3. Employ gradient accumulation.
4. Increase the number of GPUs to accommodate larger batch sizes.

**Comment by piqiuni** [comment]
@DevLinyan  Thanks for the reply!

**Comment by yuxr-smile** [comment]
> RTX4090*2

Hello, for the DriveLM project, I would like to ask how you run inference with RTX 4090 *2 GPUs. I followed the steps provided by the authors for configuration and set the batch size to 1, but I am encountering out-of-memory issues.Looking forward to your reply.
<img width="844" alt="微信图片_20241203232804" src="https://github.com/user-attachments/assets/00ccea88-08b1-4857-86af-f1f9aa03a8a3">
